sunxspex/emission.py

```diff
@@ -0,0 +1,814 @@
+    P2 = np.where(photon_energies < eebrk)[0]
+    P3 = np.where(photon_energies <= eehigh)[0]
+
+    # Part 1, below en_val[0] (usually eelow)
+    if model == 'thick-target':
+        if P1.size > 0:
+            print('Part1')
+            a_lg = np.log10(photon_energies[P1])
+            b_lg = np.log10(np.full_like(a_lg, eelow))
+            i = np.copy(P1)
+            intsum1, ier1 = integrate_part(model=model, maxfcn=maxfcn, rerr=rerr,
+                                           photon_energies=photon_energies,
+                                           eelow=eelow, eebrk=eebrk, eehigh=eehigh, p=p, q=q, z=z,
+                                           a_lg=a_lg, b_lg=b_lg, ll=i, efd=efd)
+
+            # ier = 1 indicates no convergence.
+            if sum(ier1) > 0:
+                raise ValueError('Part 1 integral did not converge for some photon energies.')
+
+    # Part 2, between enval[0] and en_val[1](usually eelow and eebrk)
+
+    aa = np.copy(photon_energies)
+    if (P2.size > 0) and (eebrk > eelow):
+        # TODO check if necessary as integration should only be carried out over point P2 which
+        # by definition are not in P1
+        if P1.size > 0:
+            aa[P1] = eelow
+
+        print('Part2')
+        a_lg = np.log10(aa[P2])
+        b_lg = np.log10(np.full_like(a_lg, eebrk))
+        i = np.copy(P2)
+        intsum2, ier2 = integrate_part(model=model, maxfcn=maxfcn, rerr=rerr,
+                                       photon_energies=photon_energies,
+                                       eelow=eelow, eebrk=eebrk, eehigh=eehigh, p=p, q=q, z=z,
+                                       a_lg=a_lg, b_lg=b_lg, ll=i, efd=efd)
+
+        if sum(ier2) > 0:
+            raise ValueError('Part 2 integral did not converge for some photon energies.')
+
+    # Part 3: between eebrk and eehigh(usually eebrk and eehigh)
+    aa = np.copy(photon_energies)
+    if (P3.sum() > 0) and (eehigh > eebrk):
+        if P2.size > 0:
+            aa[P2] = eebrk
+
+        print('Part3')
+        a_lg = np.log10(aa[P3])
+        b_lg = np.log10(np.full_like(a_lg, eehigh))
+        i = np.copy(P3)
+        intsum3, ier3 = integrate_part(model=model, maxfcn=maxfcn, rerr=rerr,
+                                       photon_energies=photon_energies,
+                                       eelow=eelow, eebrk=eebrk, eehigh=eehigh, p=p, q=q, z=z,
+                                       a_lg=a_lg, b_lg=b_lg, ll=i, efd=efd)
+        if sum(ier3) > 0:
+            raise ValueError('Part 3 integral did not converge for some photon energies.')
+
+    # TODO check units here
+    # Combine 3 parts and convert units and return
+    if model == 'thick-target':
+        DmlinO = (intsum1 + intsum2 + intsum3) * (mc2 / clight)
+        ier = ier1 + ier2 + ier3
+        return DmlinO, ier
+    elif model == 'thin-target':
+        Dmlin = (intsum2 + intsum3)
+        ier = ier2 + ier3
+        return Dmlin, ier
```

Comment:
return outside the `if` and keep the two variables `Dmlin` and `DmlinO` the same.